src/model.rs

```diff
@@ -1,173 +1,207 @@
 use core::panic;
 use std::collections::BTreeMap;
-use std::mem::MaybeUninit;
+use std::mem::{self, MaybeUninit};
 
 use crate::constraint::Constraint;
 use crate::retcode::Retcode;
-use crate::scip_call;
 use crate::solution::Solution;
 use crate::status::Status;
 use crate::variable::{VarId, VarType, Variable};
 use crate::{ffi, scip_call_panic};
+use crate::{scip_call, scip_call_expect};
 use std::ffi::CString;
 
-#[derive(Debug, PartialEq, Eq)]
-pub enum State {
-    Created,
-    PluginsIncluded,
-    ProblemCreated,
-    Solved,
+struct ScipPtr(*mut ffi::SCIP);
+
+impl ScipPtr {
+    fn new() -> Self {
+        let mut scip_ptr = MaybeUninit::uninit();
+        scip_call_panic!(ffi::SCIPcreate(scip_ptr.as_mut_ptr()));
+        let scip_ptr = unsafe { scip_ptr.assume_init() };
+        ScipPtr(scip_ptr)
+    }
+
+    fn ptr(&self) -> *mut ffi::SCIP {
+        self.0
+    }
```

Comment:
If instead of exposing the raw pointer, you exposed scip functions as methods on this struct, you could easily guarantee that they are always called on correctly initialized memory.

Reply:
Great idea, will do.